### PR TITLE
Improve the HTTP data binding BBE

### LINKED ISSUE #19127: Improve the HTTP data binding BBE to make use of all the fields in the bound record
**Description:**
$title as mentioned in https://ballerina-platform.slack.com/archives/C47EAELR1/p1568910584038800 - 
https://github.com/ballerina-platform/ballerina-lang/blob/master/examples/http-data-binding/http_data_binding.bal#L7 is not used. 

Would be useful to demonstrate chained access using the map (or maybe we could make that a record too).

### PR DESCRIPTION
**Pull request by chamil321**
Fixes #19127

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: examples/http-data-binding/http_data_binding.bal [hunk 1]
```diff
@@ -62,8 +62,10 @@ service hello on new http:Listener(9090) {
         //Accesses the fields of the record `Student`.
```
Comment (praneesha) [inline examples/http-data-binding/http_data_binding.bal]:
```suggestion
        //Accesses the fields of the `Student` record.
```

File: examples/http-data-binding/http_data_binding.client.out [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # To invoke the `bindJson` resource, use the following HTTP request.
```
Comment (praneesha) [inline examples/http-data-binding/http_data_binding.client.out]:
```suggestion
# To invoke the `bindJson` resource, execute the below HTTP request.
```

File: examples/http-data-binding/http_data_binding.client.out [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
 $ curl -v http://localhost:9090/hello/bindJson -d '{ "Details": { "ID": "77999", "Name": "XYZ"} , "Location": { "No": "01", "City": "Colombo"}}' -H "Content-Type:application/json"
 # Server response:
-{"ID":"77999","Name":"XYZ"}
+{"ID":"77999", "Name":"XYZ"}
 
 # To invoke the `bindXML` resource, use the following  HTTP request.
```
Comment (praneesha) [inline examples/http-data-binding/http_data_binding.client.out]:
```suggestion
# To invoke the `bindXML` resource, execute the below  HTTP request.
```

File: examples/http-data-binding/http_data_binding.client.out [hunk 4]
```diff
@@ -11,4 +11,4 @@ $ curl -v http://localhost:9090/hello/bindXML -d "<h:Store id = \"AST\" xmlns:h=
 # To invoke the `bindStruct` resource, use the following  HTTP request.
```
Comment (praneesha) [inline examples/http-data-binding/http_data_binding.client.out]:
```suggestion
# To invoke the `bindStruct` resource, execute the below HTTP request.
```

File: examples/http-data-binding/http_data_binding.bal [hunk 5]
```diff
@@ -59,11 +59,13 @@ service hello on new http:Listener(9090) {
...
     resource function bindStruct(http:Caller caller, http:Request req,
                                  Student student) {
-        //Accesses the fields of the record `Student`.
+        //Accesses the fields of the `Student` record.
         string name = student.Name;
         int grade = student.Grade;
+        string english = <string> student.Marks["English"];
         http:Response res = new;
-        res.setPayload({ Name: <@untainted> name, Grade: <@untainted> grade });
+        res.setPayload({ Name: <@untainted> name, Grade: <@untainted> grade,
```
Comment (MaryamZi) [inline examples/http-data-binding/http_data_binding.bal]: Shall we break this down to one field per line?

**Pull request merged**